### DISCUSSION TITLE
generate my own configuration

### DISCUSSION THREAD

**Discussion by thomasg341** [prometheus/snmp_exporter#865] [Q&A]
Hello,

I've built the generator and I'm stuck. I've read the doc but I don't understand how to generate my own config. What I'm trying to achieve is to monitor HP PDU (power distribution units) using snmp_exporter and grafana. I've got the specific mibs. And now what? Where do I put them in order to generate a snmp.yml specific to these mibs? I don't need all the mibs downloaded using make mibs.

make generate download a lot of mibs I don't want and write a generator.yml which is useless for me.

The doc says : Most use cases should be covered by our [default configuration](https://github.com/prometheus/snmp_exporter/blob/main/snmp.yml). If you need to generate your own configuration from MIBs, you can use the [generator](https://github.com/prometheus/snmp_exporter/blob/main/generator).

Use the generator if you need to customize which objects are walked or use non-public MIBs.

But how ...

Thanks a lot!

**Comment by SuperQ** [comment]
Once you have the MIB file, you will want to put a copy in the directory where all the rest of your MIBs are. If you don't have a collection, you can get a basic one with `make mibs` in the generator dir.

The hard of SNMP is knowing which OIDs you want to walk in order to get the data you need. You will need to examine the MIB file(s) to see what OIDs are available.

Once you know what you want, you can configure a `generator.yml` file with the list.

Then you can run
```
export MIBDIRS=path/to/my/mibs
generator generate
```

**Reply by lenkoda** [reply]
That's what I thought. Ok, I'll give it a try.

Thank you
